client/app/tasks/components/DeleteTaskModal.tsx

```diff
@@ -0,0 +1,46 @@
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from '@/_components/ui'
+import { useDeleteTask } from '../hooks'
+
+type DeleteTaskModalProps = {
+  isOpen: boolean
+  onClose: (showTaskModal?: boolean) => void
+  taskId: number
+}
+
+const DeleteTaskModal = ({ isOpen, onClose, taskId }: DeleteTaskModalProps) => {
+  const { onDeleteTask, isPending } = useDeleteTask(onClose)
+
+  return (
+    <AlertDialog open={isOpen} onOpenChange={() => onClose()}>
+      <AlertDialogContent>
+        <AlertDialogHeader>
+          <AlertDialogTitle>Are you absolutely sure?</AlertDialogTitle>
+          <AlertDialogDescription>
+            This action cannot be undone. This will permanently delete the task
+            and remove the data from our servers.
+          </AlertDialogDescription>
+        </AlertDialogHeader>
+        <AlertDialogFooter>
+          <AlertDialogCancel disabled={isPending}>Cancel</AlertDialogCancel>
```

Comment:
For the better user experience, isn't it better to show the previous modal when they cancel the delete modal, just like I did in the house management modal?